File: src/AES_Client_Server/wwwroot/js/crypto.js
function Crypto() {
    // Encryption
    this.encrypt = function (toEncrypt, key256) {
        // Key (utf8) as Crypto Array
        var keyArray = CryptoJS.enc.Utf8.parse(key256);
        // Generate IV (Initialization Vector)
        var iv = CryptoJS.lib.WordArray.random(16);
        // Encryption Function
        var encrypted = CryptoJS.AES.encrypt(
            // Text to Encrypt
            toEncrypt,
            // Key
            keyArray,
            {
                // IV
                iv: iv,
                // CBC (Cipher Block Chaining) Mode
                mode: CryptoJS.mode.CBC,
                // PKCS7 Padding
                padding: CryptoJS.pad.Pkcs7
            });
        // Encrypted Text as Byte Array
        var textArray = this.toByteArray(encrypted.toString());
        // IV as Byte Array
        var ivArray = this.toByteArray(CryptoJS.enc.Base64.stringify(iv));
        // Combined Byte Array (Text and IV)
        var combined = this.appendArray(textArray, ivArray);
        // Return Encrypted String (Base-64)
        return this.toBase64(combined);
    };

    // Decryption
    this.decrypt = function (toDecrypt, key256) {
        // Input String as Byte Array
        var bytes = this.toByteArray(toDecrypt);
        // Split Byte Array (Text and IV)
        var combined = this.splitArray(bytes);
        // Byte Array Containing Encrypted Text
        var text = this.toBase64(combined[0]);
        // IV Byte Array as String
        var iv = this.toBase64(combined[1]);
        // Key (utf8) as Crypto Array
        var keyArray = CryptoJS.enc.Utf8.parse(key256);
        // Decryption Function ~ Output Decypted Text as Byte Array
        var decrypted = CryptoJS.AES.decrypt(
            {
                // Text (base64) as Crypto Array
                ciphertext: CryptoJS.enc.Base64.parse(text)
            },
            keyArray,
            {
                // IV (base64) as Crypto Array
                iv: CryptoJS.enc.Base64.parse(iv),
                // CBC (Cipher Block Chaining) Mode
                mode: CryptoJS.mode.CBC,
                // PKCS7 Padding
                padding: CryptoJS.pad.Pkcs7
            });
        // Return Decrypted (UTF8) String
        return decrypted.toString(CryptoJS.enc.Utf8);
    };

    // Append IV to Text
    this.appendArray = function (text, iv) {
        // Create New Byte Array
        var result = new Uint8Array(text.length + iv.length);
        // For Combined Length, Append Text then IV
        for (i = 0; i < result.length; i++) {
            if (i < text.length) {
                result[i] = text[i];
            } else {
                result[i] = iv[i - text.length];
            }
        }
        // Return Combined Byte Array
        return result;
    };

    // Split Text and IV
    this.splitArray = function (combined) {
        // Create New Result List
        var resultList = [];
        // IV Length ~~ 16 Chars
        var ivLength = 16;
        // Length of Text minus IV
        var textLength = combined.length - ivLength;
        // Copy Text to Byte Array
        var text = combined.subarray(0, textLength);
        // Copy IV to Byte Array
        var iv = combined.subarray(textLength, combined.length);
        // Add Text Byte Array to Result List
        resultList.push(text);
        // Add IV Byte Array to Result List
        resultList.push(iv);
        // Return Result List
        return resultList;
    };

    // Base64 to Byte Array
    this.toByteArray = function (base64) {
        // Decode String from Base64
        var binStr = atob(base64);
        // Create New Byte Array
        var bytes = new Uint8Array(binStr.length);
        // Assign Array Element i the CharCode Value of String Index i
        Array.prototype.forEach.call(binStr, function (ch, i) {
            bytes[i] = ch.charCodeAt(0);
        });
        // Return Byte Array
        return bytes;
    };

    // Byte Array to Base64
    this.toBase64 = function (bytes) {
        // Append Each Array Element to Next Index of New String
        var base64 = Array.prototype.map.call(bytes, function (ch) {
            return String.fromCharCode(ch);
        }).join('');
        // Return String Encoded as Base64
        return btoa(base64);
    };
};

File: Crypto.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Security.Cryptography;

namespace aes_client_server
{
    public class Crypto
    {
        // Encryption
        public static string Encrypt(string toEncrypt, string key256)
        {
            // Text as Byte Array
            byte[] toEncryptArray = Encoding.UTF8.GetBytes(toEncrypt);
            // Key as Byte Array
            byte[] keyArray = Encoding.UTF8.GetBytes(key256);
            // AES Instance Creation
            Aes aes = Aes.Create();
            // Generate IV (Initialization Vector)
            aes.GenerateIV();
            // Assign Key Array
            aes.Key = keyArray;
            // CBC (Cipher Block Chaining) Mode
            aes.Mode = CipherMode.CBC;
            // PKCS7 Padding
            aes.Padding = PaddingMode.PKCS7;
            // Encryptor Creation
            ICryptoTransform cTransform = aes.CreateEncryptor();
            // Encrypted Text as Byte Array
            byte[] textArray = cTransform.TransformFinalBlock(toEncryptArray, 0, toEncryptArray.Length);
            // IV as Byte Array
            byte[] ivArray = aes.IV;
            // Combined Byte Array (Text and IV)
            byte[] combinedArray = AppendArray(textArray, ivArray);
            // Return Encrypted String (Base-64)
            return Convert.ToBase64String(combinedArray, 0, combinedArray.Length);
        }

        // Decryption
        public static string Decrypt(string toDecrypt, string key256)
        {
            // Input String as Byte Array
            byte[] bytes = Convert.FromBase64String(toDecrypt);
            // Split Byte Array (Text and IV)
            List<byte[]> combined = SplitArray(bytes);
            // Byte Array Containing Encrypted Text
            byte[] toDecryptArray = combined[0];
            // IV as Byte Array
            byte[] iv = combined[1];
            // Key as Byte Array
            byte[] keyArray = Encoding.UTF8.GetBytes(key256);
            // AES Instance Creation
            Aes aes = Aes.Create();
            // Assign IV
            aes.IV = iv;
            // Assign Key Array
            aes.Key = keyArray;
            // CBC (Cipher Block Chaining) Mode
            aes.Mode = CipherMode.CBC;
            // PKCS7 Padding
            aes.Padding = PaddingMode.PKCS7;
            // Decryptor Creation
            ICryptoTransform cTransform = aes.CreateDecryptor();
            // Output Decypted Text as Byte Array
            byte[] resultArray = cTransform.TransformFinalBlock(toDecryptArray, 0, toDecryptArray.Length);
            // Return Decrypted (UTF8) String
            return Encoding.UTF8.GetString(resultArray);
        }

        // Append IV to Text
        private static byte[] AppendArray(byte[] text, byte[] iv)
        {
            // Create New Byte Array
            byte[] result = new byte[text.Length + iv.Length];
            // Add Text Byte Array to 'result' Starting at Index 0
            Array.Copy(text, 0, result, 0, text.Length);
            // Add IV Byte Array to 'result' where Text Ends
            Array.Copy(iv, 0, result, text.Length, iv.Length);
            // Return Combined Byte Array
            return result;
        }

        // Split Text and IV
        private static List<byte[]> SplitArray(byte[] combined)
        {
            // Create New Result List
            List<byte[]> resultList = new List<byte[]>();
            // IV Length ~~ 16 Chars
            int ivLength = 16;
            // Length of Text minus IV
            int textLength = combined.Length - ivLength;
            // Create 2 New Byte Arrays
            byte[] text = new byte[textLength];
            byte[] iv = new byte[ivLength];
            // Copy Text to Byte Array
            Array.Copy(combined, 0, text, 0, textLength);
            // Copy IV to Byte Array
            Array.Copy(combined, textLength, iv, 0, ivLength);
            // Add Text Byte Array to Result List
            resultList.Add(text);
            // Add IV Byte Array to Result List
            resultList.Add(iv);
            // Return Result List
            return resultList;
        }
    }
}
